src/NUnitFramework/framework/Internal/Results/TestResult.cs

```diff
@@ -645,10 +645,10 @@ private TNode AddFailureElement(TNode targetNode)
         {
             TNode failureNode = targetNode.AddElement("failure");
 
-            if (Message != null)
+            if (Message != null && Message.Trim().Length > 0)
                 failureNode.AddElementWithCDATA("message", Message);
 
-            if (StackTrace != null)
+            if (StackTrace != null && StackTrace.Trim().Length > 0)
                 failureNode.AddElementWithCDATA("stack-trace", StackTrace);
 
             return failureNode;
```

Comment:
In theory, we could end up with an empty failure element, although it's not likely. We have to keep the element in any case because people often collect failed tests by looking for that element.